### DISCUSSION TITLE
"function not implemented" error starting AGH

### DISCUSSION THREAD

**Discussion by neilio** [AdguardTeam/AdGuardHome#4814] [Q&A]
Hey folks - I've been running AGH on a Synology DS413j for the past six months as both DNS and DHCP with no issues. I upgraded manually during that time and had no issues.

Today when I tried to manually upgrade, launching AGH gives me the error:

`accept tcp [::]:8888: accept: function not implemented` 

I'm not clear why this is happening and downgrading to the previous version causes the same error as well, so I assume I need to fix something on the NAS to get it working. Can someone point me in the right direction?

AdGuard Home, version v0.107.9
Using AdGuardHome_linux_armv5.tar.gz

**Comment by neilio** [comment]
For anyone who searches for this – I had to downgrade to version 0.107.8 to get it working again.

**Comment by ainar-g** [comment]
Hello, and apologies for the long silence.  We have fixed a similar issue not long ago.  Could you please try one of the latest releases and see if the error message is still there?  If it is, please file an issue with all the details.  Thanks!

**Comment by smartmic** [comment]
Hi, I have a similar issue.
I wanted to use AdGuardHome (version 0.107.19) on my Asus Router RT-N66U, an older box (~9 yrs) running Asuswrt-Merlin:

    # uname -a
    Linux RT-N66U 2.6.22.19 #1 Fri Nov 11 16:42:24 EST 2016 mips GNU/Linux

Here is my output:
```
# ./AdGuardHome
2022/11/30 22:19:42.956892 [info] AdGuard Home, version v0.107.19
2022/11/30 22:19:42.957475 [info] This is the first time AdGuard Home is launched
2022/11/30 22:19:42.957674 [info] Checking if AdGuard Home has necessary permissions
2022/11/30 22:19:42.957929 [info] warning: cannot check capability cap_net_bind_service: invalid argument
2022/11/30 22:19:42.959027 [info] AdGuard failed to bind to port 53: listen tcp 127.0.0.1:53: bind: address already in use

Please note, that this is crucial for a DNS server to be able to use that port.
2022/11/30 22:19:42.959243 [info] AdGuard Home can bind to port 53
2022/11/30 22:19:42.963318 [error] creating dhcpv4 srv: dhcpv4: invalid IP is not an IPv4 address
2022/11/30 22:19:42.964931 [info] Initializing auth module: /tmp/mnt/sda1/local/AdGuardHome/data/sessions.db
2022/11/30 22:19:42.965710 [info] auth: initialized.  users:0  sessions:0
2022/11/30 22:19:42.965988 [info] web: initializing
2022/11/30 22:19:42.966300 [info] This is the first launch of AdGuard Home, redirecting everything to /install.html
2022/11/30 22:19:42.966646 [info] AdGuard Home is available at the following addresses:
2022/11/30 22:19:42.969595 [info] Go to http://127.0.0.1:3000
2022/11/30 22:19:42.969832 [info] Go to http://192.168.0.20:3000
2022/11/30 22:19:42.970041 [info] Go to http://192.168.1.1:3000
2022/11/30 22:19:42.971228 [info] stopped
2022/11/30 22:19:42.971510 [fatal] accept tcp [::]:3000: accept: function not implemented
```

Any hope for me to get AGH running there?
